manim/utils/color/core.py

```diff
@@ -536,10 +598,51 @@ def interpolate(self, other: ManimColor, alpha: float) -> ManimColor:
         ManimColor
             The interpolated ManimColor
         """
-        return ManimColor(
-            self._internal_value * (1 - alpha) + other._internal_value * alpha
+        return self._construct_from_space(
+            self._internal_space * (1 - alpha) + other._internal_space * alpha
         )
 
+    def opacity(self, opacity: float) -> Self:
+        """Creates a new ManimColor with the given opacity and the same color value as before
+
+        Parameters
+        ----------
+        opacity : float
+            The new opacity value to be used
+
+        Returns
+        -------
+        ManimColor
+            The new ManimColor with the same color value but the new opacity
+        """
+        tmp = self._internal_space.copy()
+        tmp[-1] = opacity
+        return self._construct_from_space(tmp)
+
+    def into(self, classtype: type[ManimColorT]) -> ManimColorT:
+        """Converts the current color into a different colorspace that is given without changing the _internal_value
+
+        Parameters
+        ----------
+        classtype : type[ManimColorT]
+            The class that is used for conversion, it must be a subclass of ManimColor which respects the specification
+            HSV, RGBA, ...
+
+        Returns
+        -------
+        __INTO
+            Color object of the type passed into classtype with the same internal value as previously
+        """
+        return classtype._from_internal(self._internal_value)
+
+    @classmethod
+    def _from_internal(cls, value: ManimColorInternal) -> Self:
+        """This function is intended to be overwritten by custom color space classes which are subtypes of ManimColor
+        The function constructs a new Object of the given class by transforming the value in the internal format [r,g,b,a]
+        into a format which the constructor of the custom class can understand, look at HSV for an example
```

Comment:
```suggestion
        """This function is intended to be overwritten by custom color space classes which are subtypes of ManimColor.
        
        The function constructs a new object of the given class by transforming the value in the internal format ``[r,g,b,a]``
        into a format which the constructor of the custom class can understand. Look at :class:`.HSV` for an example.
```
This way it looks less weirdly formatted in the docs.
![image](https://github.com/user-attachments/assets/527ae75e-2bc7-44b1-9359-98413c0f71b1)